Modules/_io/textio.c

```diff
@@ -1112,6 +1112,10 @@ _io_TextIOWrapper___init___impl(textio *self, PyObject *buffer,
     else if (io_check_errors(errors)) {
         return -1;
     }
+    const char *errors_str = PyUnicode_AsUTF8(errors);
```

Comment:
I suggest to reject embedded null characters:

```suggestion
    const char *errors_str = PyUnicode_AsUTF8NoNUL(errors);
```